github/GithubObject.py

```diff
@@ -63,15 +63,16 @@
     from .Requester import Requester
 
 T = typing.TypeVar("T")
+T_co = typing.TypeVar("T_co", covariant=True)
 
 
-class Attribute(Generic[T]):
+class Attribute(Protocol[T_co]):
     @property
-    def value(self) -> T:
+    def value(self) -> T_co:
         raise NotImplementedError
 
 
-class _NotSetType(Attribute):
+class _NotSetType:
```

Comment:
I think

    self._id: Attribute[int] = NotSet

is not allowed any more, because `NotSet` is not an `Attribute` any more.

Reply:
no

`typing.Protocol` is special. It's duck type checking, not subclass checking

Reply:
Mypy check passed